drivers/storage/ide/pciidex/pdo.c

```diff
@@ -428,8 +428,7 @@ PciIdeXPdoPnpDispatch(
 				{
 					DPRINT1("IRP_MJ_PNP / IRP_MN_QUERY_DEVICE_RELATIONS / Unknown type 0x%lx\n",
 						Stack->Parameters.QueryDeviceRelations.Type);
-					ASSERT(FALSE);
-					Status = STATUS_NOT_SUPPORTED;
+					Status = Irp->IoStatus.Status;
```

Comment:
These lines look rather weird to me... Was Irp->IoStatus.Status initiliazed before?
Wouldn't it be better to set Status = STATUS_NOT_SUPPORTED;
Irp->IoStatus.Status = Status;?

Reply:
No, the PNP manager initializes it (to STATUS_NOT_SUPPORTED), and a higher-level driver may have set it to another status, which we must keep. The rule is: if you don't handle the specific minor function, don't touch IoStatus.
https://docs.microsoft.com/en-us/windows-hardware/drivers/kernel/dispatchpnp-routines says this somewhat indirectly, but Driver Verifier complains if you make any change to it.

Reply:
OK, thanks for the details :-).
LGTM then!